pkg/cluster/samples.go

```diff
@@ -27,7 +28,26 @@ func (m *manager) disableSamples(ctx context.Context) error {
 		},
 		func() error {
 			c, err := m.samplescli.SamplesV1().Configs().Get(ctx, "cluster", metav1.GetOptions{})
-			if err != nil {
+
+			// if config cr is missing, we create it with managementState: Removed to
+			// immediately disable the samples operator without waiting for it to
+			// bootstrap and create the resource
+			//
+			// https://docs.openshift.com/container-platform/4.17/openshift_images/configuring-samples-operator.html#samples-operator-restricted-network-install-with-access
+			if errors.IsNotFound(err) {
```

Comment:
Should we make sure that if this fails with a conflict (e.g. it gets bootstrapped, maybe?) then we leave it alone? Or retry, maybe by using clienthelper's Ensure or something?

Reply:
We already have been retrying on conflicts: https://github.com/Azure/ARO-RP/blob/23600d9a78319c232e7a9e90b39f32264a03f750/pkg/cluster/samples.go#L24.

Reply:
Oh, I guess the whole thing being wrapped in it would do it as well. I'm a derp :)